packages/react-components/src/hooks/requestFunctions/useIoTSiteWiseClient.ts

```diff
@@ -0,0 +1,63 @@
+import { IoTSiteWise, IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
+import { useMemo } from 'react';
+import {
+  createBatchGetAssetPropertyValue,
+  createBatchGetAssetPropertyValueHistory,
+  createDescribeAsset,
+  createDescribeAssetModel,
+  createExecuteQuery,
+  createGetAssetPropertyValue,
+  createGetAssetPropertyValueHistory,
+  createListAssetModelProperties,
+  createListAssetModels,
+  createListAssetProperties,
+  createListAssets,
+  createListAssociatedAssets,
+  createListTimeSeries,
+} from './data/iotsitewise';
+
+/**
+ * Hook interface accepts any AWS SDK client type for IoT SiteWise
+ */
+export interface UseIoTSiteWiseClientOptions {
+  iotSiteWiseClient?: IoTSiteWiseClient | IoTSiteWise;
+}
+
+/**
+ *
+ * @param iotSiteWiseClient is an AWS SDK client type for IoT SiteWise
+ * @returns an IoTSiteWise type, promise pattern client
+ *
+ * @experimental Do not use in production.
+ */
+export function useIoTSiteWiseClient({
```

Comment:
just to understand better, for any Sitewise or Events operation called withint react-components, we would need to add a requestFunction implementation of the operation and return in the hooks ?

Also would not be better to move `useIoTSiteWiseClient` and `useIoTEventsClient` to `core-utils` instead ? so we can use in other packages if necessary.

Reply:
The team decided offline that we'd put this in `react-components` for now. I was considering putting it in `core` or `core-utils`, but we don't want to make this part of the public API yet. We have more to discuss as a team about the direction of these request functions, and don't want that to block us.

Reply:
if we dont want to make this part of the public API yep makes sense to be in react-components, also considering that it will be only used in the react-components scope